File: backend/models/meta.py
"""Operational API contracts."""

from typing import Literal

from pydantic import BaseModel


class ReadinessResponse(BaseModel):
    status: Literal["ready", "not_ready"]
    service: str = "clearpath-api"
    environment: str
    release: str
    checks: dict[str, bool]
    station_count: int
    fresh_station_count: int
    latest_recorded_at: str | None = None
    reason: str | None = None

File: frontend/types/meta.ts
export interface ReadinessResponse {
  status: "ready" | "not_ready";
  service: "clearpath-api";
  environment: string;
  release: string;
  checks: Record<string, boolean>;
  station_count: number;
  fresh_station_count: number;
  delayed_station_count: number;
  expired_station_count: number;
  fresh_max_age_minutes: number;
  surface_max_age_minutes: number;
  latest_recorded_at: string | null;
  reason: string | null;
}
